{"nodes":[{"pos":[878,982],"content":"The exception that is thrown when SQL Server returns a warning or error. This class cannot be inherited.","needQuote":true,"needEscape":true,"nodes":[{"content":"The exception that is thrown when SQL Server returns a warning or error. This class cannot be inherited.","pos":[0,104],"nodes":[{"content":"The exception that is thrown when SQL Server returns a warning or error.","pos":[0,72]},{"content":"This class cannot be inherited.","pos":[73,104]}]}]},{"pos":[995,4971],"content":"This class is created whenever the .NET Framework Data Provider for SQL Server encounters an error generated from the server. (Client side errors are thrown as standard common language runtime exceptions.) SqlException always contains at least one instance of <xref:System.Data.SqlClient.SqlError>.       Messages that have a severity level of 10 or less are informational and indicate problems caused by mistakes in information that a user has entered. Severity levels from 11 through 16 are generated by the user, and can be corrected by the user. Severity levels from 17 through 25 indicate software or hardware errors. When a level 17, 18, or 19 error occurs, you can continue working, although you might not be able to execute a particular statement.       The <xref:System.Data.SqlClient.SqlConnection> remains open when the severity level is 19 or less. When the severity level is 20 or greater, the server ordinarily closes the <xref:System.Data.SqlClient.SqlConnection>. However, the user can reopen the connection and continue. In both cases, a SqlException is generated by the method executing the command.       For information about the warning and informational messages sent by SQL Server, see the Troubleshooting section of SQL Server Books Online. The SqlException class maps to SQL Server severity.       The following is general information on handling exceptions. Your code should catch exceptions to prevent the application from crashing and to allow displaying a relevant error message to the user.  You can use database transactions to ensure that the data is consistent regardless of what happens in the client application (including a crash). Features like System.Transaction.TransactionScope or the BeginTransaction method (in System.Data.OleDb.OleDbConnection, System.Data.ODBC.ODBCConnection, and System.Data.SqlClient.SqlConnection) ensure consistent data regardless of exceptions raised by a provider. Transactions can fail, so catch failures and retry the transaction.       Note that beginning with  .NET Framework 4.5, SqlException can return an inner <xref:System.ComponentModel.Win32Exception>.       The exception class of a .Net Framework data provider reports provider-specific errors. For example System.Data.Odbc has OdbcException, System.Data.OleDb has OleDbException, and System.Data.SqlClient has SqlException. For the best level of error detail, catch these exceptions and use the members of these exception classes to get details of the error.       In addition to the provider-specific errors, .NET Framework data provider types can raise .NET Framework exceptions such as System.OutOfMemoryException and System.Threading.ThreadAbortException. Recovery from these exceptions may not be possible.       Bad input can cause a .NET Framework data provider type to raise an exception such as System.ArgumentException or System.IndexOutOfRangeException.  Calling a method at the wrong time can raise System.InvalidOperationException.       So, in general, write an exception handler that catches any provider specific exceptions as well as exceptions from the common language runtime.  These can be layered as follows:      ```   Try {      // code here   }   catch (SqlException odbcEx) {      // Handle more specific SqlException exception here.   }   catch (Exception ex) {      // Handle generic ones here.   }      ```       Or:      ```   Try {      // code here   }   catch (Exception ex) {      if (ex is SqlException) {         // Handle more specific SqlException exception here.      }      Else {         // Handle generic ones here.      }   }      ```       It is also possible for a .NET Framework data provider method call to fail on a thread pool thread with no user code on the stack. In this case, and when using asynchronous method calls, you must register the <xref:System.AppDomain.UnhandledException> event to handle those exceptions and avoid application crash.","needQuote":false,"needEscape":true,"nodes":[{"content":"This class is created whenever the .NET Framework Data Provider for SQL Server encounters an error generated from the server.","pos":[0,125]},{"content":"(Client side errors are thrown as standard common language runtime exceptions.) SqlException always contains at least one instance of &lt;xref:System.Data.SqlClient.SqlError&gt;.","pos":[126,298],"source":" (Client side errors are thrown as standard common language runtime exceptions.) SqlException always contains at least one instance of <xref:System.Data.SqlClient.SqlError>."},{"content":"Messages that have a severity level of 10 or less are informational and indicate problems caused by mistakes in information that a user has entered.","pos":[305,453]},{"content":"Severity levels from 11 through 16 are generated by the user, and can be corrected by the user.","pos":[454,549]},{"content":"Severity levels from 17 through 25 indicate software or hardware errors.","pos":[550,622]},{"content":"When a level 17, 18, or 19 error occurs, you can continue working, although you might not be able to execute a particular statement.","pos":[623,755]},{"content":"The &lt;xref:System.Data.SqlClient.SqlConnection&gt; remains open when the severity level is 19 or less.","pos":[762,860],"source":"       The <xref:System.Data.SqlClient.SqlConnection> remains open when the severity level is 19 or less."},{"content":"When the severity level is 20 or greater, the server ordinarily closes the &lt;xref:System.Data.SqlClient.SqlConnection&gt;.","pos":[861,979],"source":" When the severity level is 20 or greater, the server ordinarily closes the <xref:System.Data.SqlClient.SqlConnection>."},{"content":"However, the user can reopen the connection and continue.","pos":[980,1037]},{"content":"In both cases, a SqlException is generated by the method executing the command.","pos":[1038,1117]},{"content":"For information about the warning and informational messages sent by SQL Server, see the Troubleshooting section of SQL Server Books Online.","pos":[1124,1264]},{"content":"The SqlException class maps to SQL Server severity.","pos":[1265,1316]},{"content":"The following is general information on handling exceptions.","pos":[1323,1383]},{"content":"Your code should catch exceptions to prevent the application from crashing and to allow displaying a relevant error message to the user.","pos":[1384,1520]},{"content":"You can use database transactions to ensure that the data is consistent regardless of what happens in the client application (including a crash).","pos":[1522,1667]},{"content":"Features like System.Transaction.TransactionScope or the BeginTransaction method (in System.Data.OleDb.OleDbConnection, System.Data.ODBC.ODBCConnection, and System.Data.SqlClient.SqlConnection) ensure consistent data regardless of exceptions raised by a provider.","pos":[1668,1931]},{"content":"Transactions can fail, so catch failures and retry the transaction.","pos":[1932,1999]},{"content":"Note that beginning with  .NET Framework 4.5, SqlException can return an inner &lt;xref:System.ComponentModel.Win32Exception&gt;.","pos":[2006,2129],"source":"       Note that beginning with  .NET Framework 4.5, SqlException can return an inner <xref:System.ComponentModel.Win32Exception>."},{"content":"The exception class of a .Net Framework data provider reports provider-specific errors.","pos":[2136,2223]},{"content":"For example System.Data.Odbc has OdbcException, System.Data.OleDb has OleDbException, and System.Data.SqlClient has SqlException.","pos":[2224,2353]},{"content":"For the best level of error detail, catch these exceptions and use the members of these exception classes to get details of the error.","pos":[2354,2488]},{"content":"In addition to the provider-specific errors, .NET Framework data provider types can raise .NET Framework exceptions such as System.OutOfMemoryException and System.Threading.ThreadAbortException.","pos":[2495,2689]},{"content":"Recovery from these exceptions may not be possible.","pos":[2690,2741]},{"content":"Bad input can cause a .NET Framework data provider type to raise an exception such as System.ArgumentException or System.IndexOutOfRangeException.","pos":[2748,2894]},{"content":"Calling a method at the wrong time can raise System.InvalidOperationException.","pos":[2896,2974]},{"content":"So, in general, write an exception handler that catches any provider specific exceptions as well as exceptions from the common language runtime.","pos":[2981,3125]},{"content":"These can be layered as follows:      <ph id=\"ph1\">```   Try {      // code here   }   catch (SqlException odbcEx) {      // Handle more specific SqlException exception here.   }   catch (Exception ex) {      // Handle generic ones here.   }      ```</ph>       Or:      <ph id=\"ph2\">```   Try {      // code here   }   catch (Exception ex) {      if (ex is SqlException) {         // Handle more specific SqlException exception here.      }      Else {         // Handle generic ones here.      }   }      ```</ph>       It is also possible for a .NET Framework data provider method call to fail on a thread pool thread with no user code on the stack.","pos":[3127,3743],"source":"  These can be layered as follows:      ```   Try {      // code here   }   catch (SqlException odbcEx) {      // Handle more specific SqlException exception here.   }   catch (Exception ex) {      // Handle generic ones here.   }      ```       Or:      ```   Try {      // code here   }   catch (Exception ex) {      if (ex is SqlException) {         // Handle more specific SqlException exception here.      }      Else {         // Handle generic ones here.      }   }      ```       It is also possible for a .NET Framework data provider method call to fail on a thread pool thread with no user code on the stack."},{"content":"In this case, and when using asynchronous method calls, you must register the &lt;xref:System.AppDomain.UnhandledException&gt; event to handle those exceptions and avoid application crash.","pos":[3744,3926],"source":" In this case, and when using asynchronous method calls, you must register the <xref:System.AppDomain.UnhandledException> event to handle those exceptions and avoid application crash."}]},{"pos":[5988,6087],"content":"Gets the severity level of the error returned from the .NET Framework Data Provider for SQL Server.","needQuote":true,"needEscape":true,"nodes":[{"content":"Gets the severity level of the error returned from the .NET Framework Data Provider for SQL Server.","pos":[0,99]}]},{"pos":[6100,7307],"content":"Messages that have a severity level of 10 or less are informational and indicate problems caused by mistakes in information that a user has entered. Severity levels from 11 through 16 are generated by the user, and can be corrected by the user. Severity levels from 17 through 25 indicate software or hardware errors. When a level 17, 18, or 19 error occurs, you can continue working, although you might not be able to execute a particular statement.       The <xref:System.Data.SqlClient.SqlConnection> remains open when the severity level is 19 or less. When the severity level is 20 or greater, the server ordinarily closes the <xref:System.Data.SqlClient.SqlConnection>. However, the user can reopen the connection and continue. In both cases, a <xref:System.Data.SqlClient.SqlException> is generated by the method executing the command.       For information about the warning and informational messages sent by SQL Server, see the Troubleshooting section of the SQL Server documentation.       This is a wrapper for the <xref:System.Data.SqlClient.SqlError.Class%2A> property of the first <xref:System.Data.SqlClient.SqlError> in the <xref:System.Data.SqlClient.SqlException.Errors%2A> property.","needQuote":false,"needEscape":true,"nodes":[{"content":"Messages that have a severity level of 10 or less are informational and indicate problems caused by mistakes in information that a user has entered. Severity levels from 11 through 16 are generated by the user, and can be corrected by the user. Severity levels from 17 through 25 indicate software or hardware errors. When a level 17, 18, or 19 error occurs, you can continue working, although you might not be able to execute a particular statement.       The <xref:System.Data.SqlClient.SqlConnection> remains open when the severity level is 19 or less. When the severity level is 20 or greater, the server ordinarily closes the <xref:System.Data.SqlClient.SqlConnection>. However, the user can reopen the connection and continue. In both cases, a <xref:System.Data.SqlClient.SqlException> is generated by the method executing the command.       For information about the warning and informational messages sent by SQL Server, see the Troubleshooting section of the SQL Server documentation.       This is a wrapper for the <xref:System.Data.SqlClient.SqlError.Class%2A> property of the first <xref:System.Data.SqlClient.SqlError> in the <xref:System.Data.SqlClient.SqlException.Errors%2A> property.","pos":[0,1201],"nodes":[{"content":"Messages that have a severity level of 10 or less are informational and indicate problems caused by mistakes in information that a user has entered.","pos":[0,148]},{"content":"Severity levels from 11 through 16 are generated by the user, and can be corrected by the user.","pos":[149,244]},{"content":"Severity levels from 17 through 25 indicate software or hardware errors.","pos":[245,317]},{"content":"When a level 17, 18, or 19 error occurs, you can continue working, although you might not be able to execute a particular statement.","pos":[318,450]},{"content":"The &lt;xref:System.Data.SqlClient.SqlConnection&gt; remains open when the severity level is 19 or less.","pos":[457,555],"source":"       The <xref:System.Data.SqlClient.SqlConnection> remains open when the severity level is 19 or less."},{"content":"When the severity level is 20 or greater, the server ordinarily closes the &lt;xref:System.Data.SqlClient.SqlConnection&gt;.","pos":[556,674],"source":" When the severity level is 20 or greater, the server ordinarily closes the <xref:System.Data.SqlClient.SqlConnection>."},{"content":"However, the user can reopen the connection and continue.","pos":[675,732]},{"content":"In both cases, a &lt;xref:System.Data.SqlClient.SqlException&gt; is generated by the method executing the command.","pos":[733,841],"source":" In both cases, a <xref:System.Data.SqlClient.SqlException> is generated by the method executing the command."},{"content":"For information about the warning and informational messages sent by SQL Server, see the Troubleshooting section of the SQL Server documentation.","pos":[848,993]},{"content":"This is a wrapper for the &lt;xref:System.Data.SqlClient.SqlError.Class%2A&gt; property of the first &lt;xref:System.Data.SqlClient.SqlError&gt; in the &lt;xref:System.Data.SqlClient.SqlException.Errors%2A&gt; property.","pos":[1000,1201],"source":"       This is a wrapper for the <xref:System.Data.SqlClient.SqlError.Class%2A> property of the first <xref:System.Data.SqlClient.SqlError> in the <xref:System.Data.SqlClient.SqlException.Errors%2A> property."}]}]},{"pos":[7823,7891],"content":"A value from 1 to 25 that indicates the severity level of the error.","needQuote":true,"needEscape":true,"nodes":[{"content":"A value from 1 to 25 that indicates the severity level of the error.","pos":[0,68]}]},{"pos":[8371,8499],"content":"Represents the client connection ID. For more information, see [Data Tracing in ADO.NET](~/add/includes/ajax-current-ext-md.md).","needQuote":true,"needEscape":true,"nodes":[{"content":"Represents the client connection ID.","pos":[0,36]},{"content":"For more information, see <bpt id=\"p1\">[</bpt>Data Tracing in ADO.NET<ept id=\"p1\">](~/add/includes/ajax-current-ext-md.md)</ept>.","pos":[37,128],"source":" For more information, see [Data Tracing in ADO.NET](~/add/includes/ajax-current-ext-md.md)."}]},{"pos":[8511,8588],"content":"For a code sample, see <xref:System.Data.SqlClient.SqlException.ToString%2A>.","needQuote":true,"needEscape":true,"nodes":[{"content":"For a code sample, see &lt;xref:System.Data.SqlClient.SqlException.ToString%2A&gt;.","pos":[0,77],"source":"For a code sample, see <xref:System.Data.SqlClient.SqlException.ToString%2A>."}]},{"pos":[8707,8740],"content":"Returns the client connection ID.","needQuote":true,"needEscape":true,"nodes":[{"content":"Returns the client connection ID.","pos":[0,33]}]},{"pos":[9173,9375],"content":"Gets a collection of one or more <bpt id=\"p1\">&lt;xref href=\"System.Data.SqlClient.SqlError\"&gt;</bpt><ept id=\"p1\">&lt;/xref&gt;</ept> objects that give detailed information about exceptions generated by the .NET Framework Data Provider for SQL Server.","needQuote":true,"needEscape":true,"source":"Gets a collection of one or more <xref href=\"System.Data.SqlClient.SqlError\"></xref> objects that give detailed information about exceptions generated by the .NET Framework Data Provider for SQL Server."},{"pos":[9388,9693],"content":"The <xref:System.Data.SqlClient.SqlErrorCollection> class always contains at least one instance of the <xref:System.Data.SqlClient.SqlError> class.       This is a wrapper for <xref:System.Data.SqlClient.SqlErrorCollection>. For more information on SQL Server engine errors, see SQL Server Books Online.","needQuote":false,"needEscape":true,"nodes":[{"content":"The <xref:System.Data.SqlClient.SqlErrorCollection> class always contains at least one instance of the <xref:System.Data.SqlClient.SqlError> class.       This is a wrapper for <xref:System.Data.SqlClient.SqlErrorCollection>. For more information on SQL Server engine errors, see SQL Server Books Online.","pos":[0,303],"nodes":[{"content":"The &lt;xref:System.Data.SqlClient.SqlErrorCollection&gt; class always contains at least one instance of the &lt;xref:System.Data.SqlClient.SqlError&gt; class.","pos":[0,147],"source":"The <xref:System.Data.SqlClient.SqlErrorCollection> class always contains at least one instance of the <xref:System.Data.SqlClient.SqlError> class."},{"content":"This is a wrapper for &lt;xref:System.Data.SqlClient.SqlErrorCollection&gt;.","pos":[154,224],"source":"       This is a wrapper for <xref:System.Data.SqlClient.SqlErrorCollection>."},{"content":"For more information on SQL Server engine errors, see SQL Server Books Online.","pos":[225,303]}]}]},{"pos":[10255,10344],"content":"The collected instances of the <bpt id=\"p1\">&lt;xref href=\"System.Data.SqlClient.SqlError\"&gt;</bpt><ept id=\"p1\">&lt;/xref&gt;</ept> class.","needQuote":true,"needEscape":true,"source":"The collected instances of the <xref href=\"System.Data.SqlClient.SqlError\"></xref> class."},{"pos":[11094,11208],"content":"Sets the <bpt id=\"p1\">&lt;xref href=\"System.Runtime.Serialization.SerializationInfo\"&gt;</bpt><ept id=\"p1\">&lt;/xref&gt;</ept> with information about the exception.","needQuote":true,"needEscape":true,"source":"Sets the <xref href=\"System.Runtime.Serialization.SerializationInfo\"></xref> with information about the exception."},{"pos":[11221,11439],"content":"`GetObjectData` sets a `SerializationInfo` with all the exception object data targeted for serialization. During deserialization, the exception is reconstituted from the `SerializationInfo` transmitted over the stream.","needQuote":false,"needEscape":true,"nodes":[{"content":"<ph id=\"ph1\">`GetObjectData`</ph> sets a <ph id=\"ph2\">`SerializationInfo`</ph> with all the exception object data targeted for serialization.","pos":[0,105],"source":"`GetObjectData` sets a `SerializationInfo` with all the exception object data targeted for serialization."},{"content":"During deserialization, the exception is reconstituted from the <ph id=\"ph1\">`SerializationInfo`</ph> transmitted over the stream.","pos":[106,218],"source":" During deserialization, the exception is reconstituted from the `SerializationInfo` transmitted over the stream."}]},{"pos":[11714,11857],"content":"The <bpt id=\"p1\">&lt;xref href=\"System.Runtime.Serialization.SerializationInfo\"&gt;</bpt><ept id=\"p1\">&lt;/xref&gt;</ept> that holds the serialized object data about the exception being thrown.","needQuote":true,"needEscape":true,"source":"The <xref href=\"System.Runtime.Serialization.SerializationInfo\"></xref> that holds the serialized object data about the exception being thrown."},{"pos":[11953,12079],"content":"The <xref:System.Runtime.Serialization.StreamingContext> that contains contextual information about the source or destination.","needQuote":true,"needEscape":true,"nodes":[{"content":"The &lt;xref:System.Runtime.Serialization.StreamingContext&gt; that contains contextual information about the source or destination.","pos":[0,126],"source":"The <xref:System.Runtime.Serialization.StreamingContext> that contains contextual information about the source or destination."}]},{"pos":[12258,12393],"content":"The <bpt id=\"p1\">&lt;code&gt;</bpt><ph id=\"ph1\">si</ph><ept id=\"p1\">&lt;/code&gt;</ept> parameter is a null reference (<bpt id=\"p2\">&lt;xref uid=\"langword_csharp_Nothing\" name=\"Nothing\" href=\"\"&gt;</bpt><ept id=\"p2\">&lt;/xref&gt;</ept> in Visual Basic).","needQuote":true,"needEscape":true,"source":"The <code>si</code> parameter is a null reference (<xref uid=\"langword_csharp_Nothing\" name=\"Nothing\" href=\"\"></xref> in Visual Basic)."},{"pos":[12762,12866],"content":"Gets the line number within the Transact-SQL command batch or stored procedure that generated the error.","needQuote":true,"needEscape":true,"nodes":[{"content":"Gets the line number within the Transact-SQL command batch or stored procedure that generated the error.","pos":[0,104]}]},{"pos":[12879,13178],"content":"The line numbering starts at 1; if 0 is returned, the line number is not applicable.       This is a wrapper for the <xref:System.Data.SqlClient.SqlError.LineNumber%2A> property of the first <xref:System.Data.SqlClient.SqlError> in the <xref:System.Data.SqlClient.SqlException.Errors%2A> property.","needQuote":false,"needEscape":true,"nodes":[{"content":"The line numbering starts at 1; if 0 is returned, the line number is not applicable.       This is a wrapper for the <xref:System.Data.SqlClient.SqlError.LineNumber%2A> property of the first <xref:System.Data.SqlClient.SqlError> in the <xref:System.Data.SqlClient.SqlException.Errors%2A> property.","pos":[0,297],"nodes":[{"content":"The line numbering starts at 1; if 0 is returned, the line number is not applicable.","pos":[0,84]},{"content":"This is a wrapper for the &lt;xref:System.Data.SqlClient.SqlError.LineNumber%2A&gt; property of the first &lt;xref:System.Data.SqlClient.SqlError&gt; in the &lt;xref:System.Data.SqlClient.SqlException.Errors%2A&gt; property.","pos":[91,297],"source":"       This is a wrapper for the <xref:System.Data.SqlClient.SqlError.LineNumber%2A> property of the first <xref:System.Data.SqlClient.SqlError> in the <xref:System.Data.SqlClient.SqlException.Errors%2A> property."}]}]},{"pos":[13699,13798],"content":"The line number within the Transact-SQL command batch or stored procedure that generated the error.","needQuote":true,"needEscape":true,"nodes":[{"content":"The line number within the Transact-SQL command batch or stored procedure that generated the error.","pos":[0,99]}]},{"pos":[14223,14271],"content":"Gets a number that identifies the type of error.","needQuote":true,"needEscape":true,"nodes":[{"content":"Gets a number that identifies the type of error.","pos":[0,48]}]},{"pos":[14283,14564],"content":"This is a wrapper for the <xref:System.Data.SqlClient.SqlError.Number%2A> property of the first <xref:System.Data.SqlClient.SqlError> in the <xref:System.Data.SqlClient.SqlException.Errors%2A> property. For more information on SQL Server engine errors, see SQL Server Books Online.","needQuote":true,"needEscape":true,"nodes":[{"content":"This is a wrapper for the <xref:System.Data.SqlClient.SqlError.Number%2A> property of the first <xref:System.Data.SqlClient.SqlError> in the <xref:System.Data.SqlClient.SqlException.Errors%2A> property. For more information on SQL Server engine errors, see SQL Server Books Online.","pos":[0,281],"nodes":[{"content":"This is a wrapper for the &lt;xref:System.Data.SqlClient.SqlError.Number%2A&gt; property of the first &lt;xref:System.Data.SqlClient.SqlError&gt; in the &lt;xref:System.Data.SqlClient.SqlException.Errors%2A&gt; property.","pos":[0,202],"source":"This is a wrapper for the <xref:System.Data.SqlClient.SqlError.Number%2A> property of the first <xref:System.Data.SqlClient.SqlError> in the <xref:System.Data.SqlClient.SqlException.Errors%2A> property."},{"content":"For more information on SQL Server engine errors, see SQL Server Books Online.","pos":[203,281]}]}]},{"pos":[15080,15125],"content":"The number that identifies the type of error.","needQuote":true,"needEscape":true,"nodes":[{"content":"The number that identifies the type of error.","pos":[0,45]}]},{"pos":[15561,15655],"content":"Gets the name of the stored procedure or remote procedure call (RPC) that generated the error.","needQuote":true,"needEscape":true,"nodes":[{"content":"Gets the name of the stored procedure or remote procedure call (RPC) that generated the error.","pos":[0,94]}]},{"pos":[15667,15872],"content":"This is a wrapper for the <xref:System.Data.SqlClient.SqlError.Procedure%2A> property of the first <xref:System.Data.SqlClient.SqlError> in the <xref:System.Data.SqlClient.SqlException.Errors%2A> property.","needQuote":true,"needEscape":true,"nodes":[{"content":"This is a wrapper for the &lt;xref:System.Data.SqlClient.SqlError.Procedure%2A&gt; property of the first &lt;xref:System.Data.SqlClient.SqlError&gt; in the &lt;xref:System.Data.SqlClient.SqlException.Errors%2A&gt; property.","pos":[0,205],"source":"This is a wrapper for the <xref:System.Data.SqlClient.SqlError.Procedure%2A> property of the first <xref:System.Data.SqlClient.SqlError> in the <xref:System.Data.SqlClient.SqlException.Errors%2A> property."}]},{"pos":[16393,16433],"content":"The name of the stored procedure or RPC.","needQuote":true,"needEscape":true,"nodes":[{"content":"The name of the stored procedure or RPC.","pos":[0,40]}]},{"pos":[16857,16954],"content":"Gets the name of the computer that is running an instance of SQL Server that generated the error.","needQuote":true,"needEscape":true,"nodes":[{"content":"Gets the name of the computer that is running an instance of SQL Server that generated the error.","pos":[0,97]}]},{"pos":[16966,17168],"content":"This is a wrapper for the <xref:System.Data.SqlClient.SqlError.Server%2A> property of the first <xref:System.Data.SqlClient.SqlError> in the <xref:System.Data.SqlClient.SqlException.Errors%2A> property.","needQuote":true,"needEscape":true,"nodes":[{"content":"This is a wrapper for the &lt;xref:System.Data.SqlClient.SqlError.Server%2A&gt; property of the first &lt;xref:System.Data.SqlClient.SqlError&gt; in the &lt;xref:System.Data.SqlClient.SqlException.Errors%2A&gt; property.","pos":[0,202],"source":"This is a wrapper for the <xref:System.Data.SqlClient.SqlError.Server%2A> property of the first <xref:System.Data.SqlClient.SqlError> in the <xref:System.Data.SqlClient.SqlException.Errors%2A> property."}]},{"pos":[17688,17747],"content":"The name of the computer running an instance of SQL Server.","needQuote":true,"needEscape":true,"nodes":[{"content":"The name of the computer running an instance of SQL Server.","pos":[0,59]}]},{"pos":[18168,18223],"content":"Gets the name of the provider that generated the error.","needQuote":true,"needEscape":true,"nodes":[{"content":"Gets the name of the provider that generated the error.","pos":[0,55]}]},{"pos":[18235,18437],"content":"This is a wrapper for the <xref:System.Data.SqlClient.SqlError.Source%2A> property of the first <xref:System.Data.SqlClient.SqlError> in the <xref:System.Data.SqlClient.SqlException.Errors%2A> property.","needQuote":true,"needEscape":true,"nodes":[{"content":"This is a wrapper for the &lt;xref:System.Data.SqlClient.SqlError.Source%2A&gt; property of the first &lt;xref:System.Data.SqlClient.SqlError&gt; in the &lt;xref:System.Data.SqlClient.SqlException.Errors%2A&gt; property.","pos":[0,202],"source":"This is a wrapper for the <xref:System.Data.SqlClient.SqlError.Source%2A> property of the first <xref:System.Data.SqlClient.SqlError> in the <xref:System.Data.SqlClient.SqlException.Errors%2A> property."}]},{"pos":[18966,19016],"content":"The name of the provider that generated the error.","needQuote":true,"needEscape":true,"nodes":[{"content":"The name of the provider that generated the error.","pos":[0,50]}]},{"pos":[19432,19619],"content":"Gets a numeric error code from SQL Server that represents an error, warning or \"no data found\" message. For more information about how to decode these values, see SQL Server Books Online.","needQuote":true,"needEscape":true,"nodes":[{"content":"Gets a numeric error code from SQL Server that represents an error, warning or \"no data found\" message. For more information about how to decode these values, see SQL Server Books Online.","pos":[0,187],"nodes":[{"content":"Gets a numeric error code from SQL Server that represents an error, warning or \"no data found\" message.","pos":[0,103]},{"content":"For more information about how to decode these values, see SQL Server Books Online.","pos":[104,187]}]}]},{"pos":[19631,19832],"content":"This is a wrapper for the <xref:System.Data.SqlClient.SqlError.State%2A> property of the first <xref:System.Data.SqlClient.SqlError> in the <xref:System.Data.SqlClient.SqlException.Errors%2A> property.","needQuote":true,"needEscape":true,"nodes":[{"content":"This is a wrapper for the &lt;xref:System.Data.SqlClient.SqlError.State%2A&gt; property of the first &lt;xref:System.Data.SqlClient.SqlError&gt; in the &lt;xref:System.Data.SqlClient.SqlException.Errors%2A&gt; property.","pos":[0,201],"source":"This is a wrapper for the <xref:System.Data.SqlClient.SqlError.State%2A> property of the first <xref:System.Data.SqlClient.SqlError> in the <xref:System.Data.SqlClient.SqlException.Errors%2A> property."}]},{"pos":[20347,20386],"content":"The number representing the error code.","needQuote":true,"needEscape":true,"nodes":[{"content":"The number representing the error code.","pos":[0,39]}]},{"pos":[20820,21057],"content":"Returns a string that represents the current <bpt id=\"p1\">&lt;xref href=\"System.Data.SqlClient.SqlException\"&gt;</bpt><ept id=\"p1\">&lt;/xref&gt;</ept> object, and includes the client connection ID (for more information, see &lt;xref:System.Data.SqlClient.SqlException.ClientConnectionId*&gt;).","needQuote":true,"needEscape":true,"source":"Returns a string that represents the current <xref href=\"System.Data.SqlClient.SqlException\"></xref> object, and includes the client connection ID (for more information, see <xref:System.Data.SqlClient.SqlException.ClientConnectionId*>)."},{"pos":[22651,22772],"content":"A string that represents the current <bpt id=\"p1\">&lt;xref href=\"System.Data.SqlClient.SqlException\"&gt;</bpt><ept id=\"p1\">&lt;/xref&gt;</ept> object.&lt;xref:System.String&gt;.","needQuote":true,"needEscape":true,"source":"A string that represents the current <xref href=\"System.Data.SqlClient.SqlException\"></xref> object.<xref:System.String>."}],"content":"### YamlMime:ManagedReference\nitems:\n- uid: System.Data.SqlClient.SqlException\n  id: SqlException\n  children:\n  - System.Data.SqlClient.SqlException.Class\n  - System.Data.SqlClient.SqlException.ClientConnectionId\n  - System.Data.SqlClient.SqlException.Errors\n  - System.Data.SqlClient.SqlException.GetObjectData(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)\n  - System.Data.SqlClient.SqlException.LineNumber\n  - System.Data.SqlClient.SqlException.Number\n  - System.Data.SqlClient.SqlException.Procedure\n  - System.Data.SqlClient.SqlException.Server\n  - System.Data.SqlClient.SqlException.Source\n  - System.Data.SqlClient.SqlException.State\n  - System.Data.SqlClient.SqlException.ToString\n  langs:\n  - csharp\n  name: SqlException\n  nameWithType: SqlException\n  fullName: System.Data.SqlClient.SqlException\n  type: Class\n  summary: The exception that is thrown when SQL Server returns a warning or error. This class cannot be inherited.\n  remarks: \"This class is created whenever the .NET Framework Data Provider for SQL Server encounters an error generated from the server. (Client side errors are thrown as standard common language runtime exceptions.) SqlException always contains at least one instance of <xref:System.Data.SqlClient.SqlError>.  \\n  \\n Messages that have a severity level of 10 or less are informational and indicate problems caused by mistakes in information that a user has entered. Severity levels from 11 through 16 are generated by the user, and can be corrected by the user. Severity levels from 17 through 25 indicate software or hardware errors. When a level 17, 18, or 19 error occurs, you can continue working, although you might not be able to execute a particular statement.  \\n  \\n The <xref:System.Data.SqlClient.SqlConnection> remains open when the severity level is 19 or less. When the severity level is 20 or greater, the server ordinarily closes the <xref:System.Data.SqlClient.SqlConnection>. However, the user can reopen the connection and continue. In both cases, a SqlException is generated by the method executing the command.  \\n  \\n For information about the warning and informational messages sent by SQL Server, see the Troubleshooting section of SQL Server Books Online. The SqlException class maps to SQL Server severity.  \\n  \\n The following is general information on handling exceptions. Your code should catch exceptions to prevent the application from crashing and to allow displaying a relevant error message to the user.  You can use database transactions to ensure that the data is consistent regardless of what happens in the client application (including a crash). Features like System.Transaction.TransactionScope or the BeginTransaction method (in System.Data.OleDb.OleDbConnection, System.Data.ODBC.ODBCConnection, and System.Data.SqlClient.SqlConnection) ensure consistent data regardless of exceptions raised by a provider. Transactions can fail, so catch failures and retry the transaction.  \\n  \\n Note that beginning with  .NET Framework 4.5, SqlException can return an inner <xref:System.ComponentModel.Win32Exception>.  \\n  \\n The exception class of a .Net Framework data provider reports provider-specific errors. For example System.Data.Odbc has OdbcException, System.Data.OleDb has OleDbException, and System.Data.SqlClient has SqlException. For the best level of error detail, catch these exceptions and use the members of these exception classes to get details of the error.  \\n  \\n In addition to the provider-specific errors, .NET Framework data provider types can raise .NET Framework exceptions such as System.OutOfMemoryException and System.Threading.ThreadAbortException. Recovery from these exceptions may not be possible.  \\n  \\n Bad input can cause a .NET Framework data provider type to raise an exception such as System.ArgumentException or System.IndexOutOfRangeException.  Calling a method at the wrong time can raise System.InvalidOperationException.  \\n  \\n So, in general, write an exception handler that catches any provider specific exceptions as well as exceptions from the common language runtime.  These can be layered as follows:  \\n  \\n```  \\nTry {  \\n   // code here  \\n}  \\ncatch (SqlException odbcEx) {  \\n   // Handle more specific SqlException exception here.  \\n}  \\ncatch (Exception ex) {  \\n   // Handle generic ones here.  \\n}  \\n  \\n```  \\n  \\n Or:  \\n  \\n```  \\nTry {  \\n   // code here  \\n}  \\ncatch (Exception ex) {  \\n   if (ex is SqlException) {  \\n      // Handle more specific SqlException exception here.  \\n   }  \\n   Else {  \\n      // Handle generic ones here.  \\n   }  \\n}  \\n  \\n```  \\n  \\n It is also possible for a .NET Framework data provider method call to fail on a thread pool thread with no user code on the stack. In this case, and when using asynchronous method calls, you must register the <xref:System.AppDomain.UnhandledException> event to handle those exceptions and avoid application crash.\"\n  example:\n  - \"The following example generates a SqlException and then displays the exception.  \\n  \\n [!code-cs[Classic WebData SqlException Example#1](~/add/codesnippet/csharp/t-system.data.sqlclient._1_1.cs)]\\n [!code-vb[Classic WebData SqlException Example#1](~/add/codesnippet/visualbasic/t-system.data.sqlclient._1_1.vb)]\"\n  syntax:\n    content: 'public sealed class SqlException : System.Data.Common.DbException'\n  inheritance:\n  - System.Exception\n  - System.SystemException\n  - System.Runtime.InteropServices.ExternalException\n  - System.Data.Common.DbException\n  implements: []\n  inheritedMembers:\n  - System.Runtime.InteropServices.ExternalException.ErrorCode\n  platform:\n  - net462\n- uid: System.Data.SqlClient.SqlException.Class\n  id: Class\n  parent: System.Data.SqlClient.SqlException\n  langs:\n  - csharp\n  name: Class\n  nameWithType: SqlException.Class\n  fullName: System.Data.SqlClient.SqlException.Class\n  type: Property\n  assemblies:\n  - System.Data\n  namespace: System.Data.SqlClient\n  summary: Gets the severity level of the error returned from the .NET Framework Data Provider for SQL Server.\n  remarks: \"Messages that have a severity level of 10 or less are informational and indicate problems caused by mistakes in information that a user has entered. Severity levels from 11 through 16 are generated by the user, and can be corrected by the user. Severity levels from 17 through 25 indicate software or hardware errors. When a level 17, 18, or 19 error occurs, you can continue working, although you might not be able to execute a particular statement.  \\n  \\n The <xref:System.Data.SqlClient.SqlConnection> remains open when the severity level is 19 or less. When the severity level is 20 or greater, the server ordinarily closes the <xref:System.Data.SqlClient.SqlConnection>. However, the user can reopen the connection and continue. In both cases, a <xref:System.Data.SqlClient.SqlException> is generated by the method executing the command.  \\n  \\n For information about the warning and informational messages sent by SQL Server, see the Troubleshooting section of the SQL Server documentation.  \\n  \\n This is a wrapper for the <xref:System.Data.SqlClient.SqlError.Class%2A> property of the first <xref:System.Data.SqlClient.SqlError> in the <xref:System.Data.SqlClient.SqlException.Errors%2A> property.\"\n  example:\n  - \"The following example displays each <xref:System.Data.SqlClient.SqlError> within the <xref:System.Data.SqlClient.SqlErrorCollection> collection.  \\n  \\n [!code-cs[Classic WebData SqlException.State Example#1](~/add/codesnippet/csharp/p-system.data.sqlclient._21_1.cs)]\\n [!code-vb[Classic WebData SqlException.State Example#1](~/add/codesnippet/visualbasic/p-system.data.sqlclient._21_1.vb)]\"\n  syntax:\n    content: public byte Class { get; }\n    return:\n      type: System.Byte\n      description: A value from 1 to 25 that indicates the severity level of the error.\n  overload: System.Data.SqlClient.SqlException.Class*\n  exceptions: []\n  platform:\n  - net462\n- uid: System.Data.SqlClient.SqlException.ClientConnectionId\n  id: ClientConnectionId\n  parent: System.Data.SqlClient.SqlException\n  langs:\n  - csharp\n  name: ClientConnectionId\n  nameWithType: SqlException.ClientConnectionId\n  fullName: System.Data.SqlClient.SqlException.ClientConnectionId\n  type: Property\n  assemblies:\n  - System.Data\n  namespace: System.Data.SqlClient\n  summary: Represents the client connection ID. For more information, see [Data Tracing in ADO.NET](~/add/includes/ajax-current-ext-md.md).\n  remarks: For a code sample, see <xref:System.Data.SqlClient.SqlException.ToString%2A>.\n  syntax:\n    content: public Guid ClientConnectionId { get; }\n    return:\n      type: System.Guid\n      description: Returns the client connection ID.\n  overload: System.Data.SqlClient.SqlException.ClientConnectionId*\n  exceptions: []\n  platform:\n  - net462\n- uid: System.Data.SqlClient.SqlException.Errors\n  id: Errors\n  parent: System.Data.SqlClient.SqlException\n  langs:\n  - csharp\n  name: Errors\n  nameWithType: SqlException.Errors\n  fullName: System.Data.SqlClient.SqlException.Errors\n  type: Property\n  assemblies:\n  - System.Data\n  namespace: System.Data.SqlClient\n  summary: Gets a collection of one or more <xref href=\"System.Data.SqlClient.SqlError\"></xref> objects that give detailed information about exceptions generated by the .NET Framework Data Provider for SQL Server.\n  remarks: \"The <xref:System.Data.SqlClient.SqlErrorCollection> class always contains at least one instance of the <xref:System.Data.SqlClient.SqlError> class.  \\n  \\n This is a wrapper for <xref:System.Data.SqlClient.SqlErrorCollection>. For more information on SQL Server engine errors, see SQL Server Books Online.\"\n  example:\n  - \"The following example displays each <xref:System.Data.SqlClient.SqlError> within the <xref:System.Data.SqlClient.SqlErrorCollection> collection.  \\n  \\n [!code-vb[Classic WebData SqlError Example#1](~/add/codesnippet/visualbasic/p-system.data.sqlclient._59_1.vb)]\\n [!code-cs[Classic WebData SqlError Example#1](~/add/codesnippet/csharp/p-system.data.sqlclient._59_1.cs)]\"\n  syntax:\n    content: public System.Data.SqlClient.SqlErrorCollection Errors { get; }\n    return:\n      type: System.Data.SqlClient.SqlErrorCollection\n      description: The collected instances of the <xref href=\"System.Data.SqlClient.SqlError\"></xref> class.\n  overload: System.Data.SqlClient.SqlException.Errors*\n  exceptions: []\n  platform:\n  - net462\n- uid: System.Data.SqlClient.SqlException.GetObjectData(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)\n  id: GetObjectData(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)\n  parent: System.Data.SqlClient.SqlException\n  langs:\n  - csharp\n  name: GetObjectData(SerializationInfo,StreamingContext)\n  nameWithType: SqlException.GetObjectData(SerializationInfo,StreamingContext)\n  fullName: System.Data.SqlClient.SqlException.GetObjectData(SerializationInfo,StreamingContext)\n  type: Method\n  assemblies:\n  - System.Data\n  namespace: System.Data.SqlClient\n  summary: Sets the <xref href=\"System.Runtime.Serialization.SerializationInfo\"></xref> with information about the exception.\n  remarks: '`GetObjectData` sets a `SerializationInfo` with all the exception object data targeted for serialization. During deserialization, the exception is reconstituted from the `SerializationInfo` transmitted over the stream.'\n  syntax:\n    content: public override void GetObjectData (System.Runtime.Serialization.SerializationInfo si, System.Runtime.Serialization.StreamingContext context);\n    parameters:\n    - id: si\n      type: System.Runtime.Serialization.SerializationInfo\n      description: The <xref href=\"System.Runtime.Serialization.SerializationInfo\"></xref> that holds the serialized object data about the exception being thrown.\n    - id: context\n      type: System.Runtime.Serialization.StreamingContext\n      description: The <xref:System.Runtime.Serialization.StreamingContext> that contains contextual information about the source or destination.\n  overload: System.Data.SqlClient.SqlException.GetObjectData*\n  exceptions:\n  - type: System.ArgumentNullException\n    commentId: T:System.ArgumentNullException\n    description: The <code>si</code> parameter is a null reference (<xref uid=\"langword_csharp_Nothing\" name=\"Nothing\" href=\"\"></xref> in Visual Basic).\n  platform:\n  - net462\n- uid: System.Data.SqlClient.SqlException.LineNumber\n  id: LineNumber\n  parent: System.Data.SqlClient.SqlException\n  langs:\n  - csharp\n  name: LineNumber\n  nameWithType: SqlException.LineNumber\n  fullName: System.Data.SqlClient.SqlException.LineNumber\n  type: Property\n  assemblies:\n  - System.Data\n  namespace: System.Data.SqlClient\n  summary: Gets the line number within the Transact-SQL command batch or stored procedure that generated the error.\n  remarks: \"The line numbering starts at 1; if 0 is returned, the line number is not applicable.  \\n  \\n This is a wrapper for the <xref:System.Data.SqlClient.SqlError.LineNumber%2A> property of the first <xref:System.Data.SqlClient.SqlError> in the <xref:System.Data.SqlClient.SqlException.Errors%2A> property.\"\n  example:\n  - \"The following example displays each <xref:System.Data.SqlClient.SqlError> within the <xref:System.Data.SqlClient.SqlErrorCollection> collection.  \\n  \\n [!code-cs[Classic WebData SqlException.State Example#1](~/add/codesnippet/csharp/p-system.data.sqlclient._10_1.cs)]\\n [!code-vb[Classic WebData SqlException.State Example#1](~/add/codesnippet/visualbasic/p-system.data.sqlclient._10_1.vb)]\"\n  syntax:\n    content: public int LineNumber { get; }\n    return:\n      type: System.Int32\n      description: The line number within the Transact-SQL command batch or stored procedure that generated the error.\n  overload: System.Data.SqlClient.SqlException.LineNumber*\n  exceptions: []\n  platform:\n  - net462\n- uid: System.Data.SqlClient.SqlException.Number\n  id: Number\n  parent: System.Data.SqlClient.SqlException\n  langs:\n  - csharp\n  name: Number\n  nameWithType: SqlException.Number\n  fullName: System.Data.SqlClient.SqlException.Number\n  type: Property\n  assemblies:\n  - System.Data\n  namespace: System.Data.SqlClient\n  summary: Gets a number that identifies the type of error.\n  remarks: This is a wrapper for the <xref:System.Data.SqlClient.SqlError.Number%2A> property of the first <xref:System.Data.SqlClient.SqlError> in the <xref:System.Data.SqlClient.SqlException.Errors%2A> property. For more information on SQL Server engine errors, see SQL Server Books Online.\n  example:\n  - \"The following example displays each <xref:System.Data.SqlClient.SqlError> within the <xref:System.Data.SqlClient.SqlErrorCollection> collection.  \\n  \\n [!code-cs[Classic WebData SqlException.State Example#1](~/add/codesnippet/csharp/p-system.data.sqlclient._13_1.cs)]\\n [!code-vb[Classic WebData SqlException.State Example#1](~/add/codesnippet/visualbasic/p-system.data.sqlclient._13_1.vb)]\"\n  syntax:\n    content: public int Number { get; }\n    return:\n      type: System.Int32\n      description: The number that identifies the type of error.\n  overload: System.Data.SqlClient.SqlException.Number*\n  exceptions: []\n  platform:\n  - net462\n- uid: System.Data.SqlClient.SqlException.Procedure\n  id: Procedure\n  parent: System.Data.SqlClient.SqlException\n  langs:\n  - csharp\n  name: Procedure\n  nameWithType: SqlException.Procedure\n  fullName: System.Data.SqlClient.SqlException.Procedure\n  type: Property\n  assemblies:\n  - System.Data\n  namespace: System.Data.SqlClient\n  summary: Gets the name of the stored procedure or remote procedure call (RPC) that generated the error.\n  remarks: This is a wrapper for the <xref:System.Data.SqlClient.SqlError.Procedure%2A> property of the first <xref:System.Data.SqlClient.SqlError> in the <xref:System.Data.SqlClient.SqlException.Errors%2A> property.\n  example:\n  - \"The following example displays each <xref:System.Data.SqlClient.SqlError> within the <xref:System.Data.SqlClient.SqlErrorCollection> collection.  \\n  \\n [!code-cs[Classic WebData SqlException.State Example#1](~/add/codesnippet/csharp/p-system.data.sqlclient._9_1.cs)]\\n [!code-vb[Classic WebData SqlException.State Example#1](~/add/codesnippet/visualbasic/p-system.data.sqlclient._9_1.vb)]\"\n  syntax:\n    content: public string Procedure { get; }\n    return:\n      type: System.String\n      description: The name of the stored procedure or RPC.\n  overload: System.Data.SqlClient.SqlException.Procedure*\n  exceptions: []\n  platform:\n  - net462\n- uid: System.Data.SqlClient.SqlException.Server\n  id: Server\n  parent: System.Data.SqlClient.SqlException\n  langs:\n  - csharp\n  name: Server\n  nameWithType: SqlException.Server\n  fullName: System.Data.SqlClient.SqlException.Server\n  type: Property\n  assemblies:\n  - System.Data\n  namespace: System.Data.SqlClient\n  summary: Gets the name of the computer that is running an instance of SQL Server that generated the error.\n  remarks: This is a wrapper for the <xref:System.Data.SqlClient.SqlError.Server%2A> property of the first <xref:System.Data.SqlClient.SqlError> in the <xref:System.Data.SqlClient.SqlException.Errors%2A> property.\n  example:\n  - \"The following example displays each <xref:System.Data.SqlClient.SqlError> within the <xref:System.Data.SqlClient.SqlErrorCollection> collection.  \\n  \\n [!code-cs[Classic WebData SqlException.State Example#1](~/add/codesnippet/csharp/p-system.data.sqlclient._25_1.cs)]\\n [!code-vb[Classic WebData SqlException.State Example#1](~/add/codesnippet/visualbasic/p-system.data.sqlclient._25_1.vb)]\"\n  syntax:\n    content: public string Server { get; }\n    return:\n      type: System.String\n      description: The name of the computer running an instance of SQL Server.\n  overload: System.Data.SqlClient.SqlException.Server*\n  exceptions: []\n  platform:\n  - net462\n- uid: System.Data.SqlClient.SqlException.Source\n  id: Source\n  parent: System.Data.SqlClient.SqlException\n  langs:\n  - csharp\n  name: Source\n  nameWithType: SqlException.Source\n  fullName: System.Data.SqlClient.SqlException.Source\n  type: Property\n  assemblies:\n  - System.Data\n  namespace: System.Data.SqlClient\n  summary: Gets the name of the provider that generated the error.\n  remarks: This is a wrapper for the <xref:System.Data.SqlClient.SqlError.Source%2A> property of the first <xref:System.Data.SqlClient.SqlError> in the <xref:System.Data.SqlClient.SqlException.Errors%2A> property.\n  example:\n  - \"The following example displays each <xref:System.Data.SqlClient.SqlError> within the <xref:System.Data.SqlClient.SqlErrorCollection> collection.  \\n  \\n [!code-cs[Classic WebData SqlException.State Example#1](~/add/codesnippet/csharp/p-system.data.sqlclient._20_1.cs)]\\n [!code-vb[Classic WebData SqlException.State Example#1](~/add/codesnippet/visualbasic/p-system.data.sqlclient._20_1.vb)]\"\n  syntax:\n    content: public override string Source { get; }\n    return:\n      type: System.String\n      description: The name of the provider that generated the error.\n  overload: System.Data.SqlClient.SqlException.Source*\n  exceptions: []\n  platform:\n  - net462\n- uid: System.Data.SqlClient.SqlException.State\n  id: State\n  parent: System.Data.SqlClient.SqlException\n  langs:\n  - csharp\n  name: State\n  nameWithType: SqlException.State\n  fullName: System.Data.SqlClient.SqlException.State\n  type: Property\n  assemblies:\n  - System.Data\n  namespace: System.Data.SqlClient\n  summary: Gets a numeric error code from SQL Server that represents an error, warning or \"no data found\" message. For more information about how to decode these values, see SQL Server Books Online.\n  remarks: This is a wrapper for the <xref:System.Data.SqlClient.SqlError.State%2A> property of the first <xref:System.Data.SqlClient.SqlError> in the <xref:System.Data.SqlClient.SqlException.Errors%2A> property.\n  example:\n  - \"The following example displays each <xref:System.Data.SqlClient.SqlError> within the <xref:System.Data.SqlClient.SqlErrorCollection> collection.  \\n  \\n [!code-cs[Classic WebData SqlException.State Example#1](~/add/codesnippet/csharp/p-system.data.sqlclient._61_1.cs)]\\n [!code-vb[Classic WebData SqlException.State Example#1](~/add/codesnippet/visualbasic/p-system.data.sqlclient._61_1.vb)]\"\n  syntax:\n    content: public byte State { get; }\n    return:\n      type: System.Byte\n      description: The number representing the error code.\n  overload: System.Data.SqlClient.SqlException.State*\n  exceptions: []\n  platform:\n  - net462\n- uid: System.Data.SqlClient.SqlException.ToString\n  id: ToString\n  parent: System.Data.SqlClient.SqlException\n  langs:\n  - csharp\n  name: ToString()\n  nameWithType: SqlException.ToString()\n  fullName: System.Data.SqlClient.SqlException.ToString()\n  type: Method\n  assemblies:\n  - System.Data\n  namespace: System.Data.SqlClient\n  summary: Returns a string that represents the current <xref href=\"System.Data.SqlClient.SqlException\"></xref> object, and includes the client connection ID (for more information, see <xref:System.Data.SqlClient.SqlException.ClientConnectionId*>).\n  remarks: ''\n  example:\n  - \"The following C# example shows how a connection attempt to a valid server but non-existent database causes a <xref:System.Data.SqlClient.SqlException>, which includes the client connection ID:  \\n  \\n```  \\nusing System.Data.SqlClient;  \\nusing System;  \\n  \\npublic class A {  \\n   public static void Main() {  \\n      SqlConnection connection = new SqlConnection();  \\n      connection.ConnectionString = \\\"Data Source=a_valid_server;Initial Catalog=Northwinda;Integrated Security=true\\\";  \\n      try {  \\n         connection.Open();  \\n      }  \\n      catch (SqlException p) {  \\n         Console.WriteLine(\\\"{0}\\\", p.ClientConnectionId);  \\n         Console.WriteLine(\\\"{0}\\\", p.ToString());  \\n      }  \\n      connection.Close();  \\n   }  \\n}  \\n```  \\n  \\n The following Visual Basic sample is functionally equivalent to the previous (C#) sample:  \\n  \\n```  \\nImports System.Data.SqlClient  \\nImports System  \\n  \\nModule Module1  \\n  \\n    Sub Main()  \\n        Dim connection As New SqlConnection()  \\n        connection.ConnectionString = \\\"Data Source=a_valid_server;Initial Catalog=Northwinda;Integrated Security=true\\\"  \\n        Try  \\n            connection.Open()  \\n        Catch p As SqlException  \\n            Console.WriteLine(\\\"{0}\\\", p.ClientConnectionId)  \\n            Console.WriteLine(\\\"{0}\\\", p.ToString())  \\n        End Try  \\n        connection.Close()  \\n    End Sub  \\n  \\nEnd Module  \\n```\"\n  syntax:\n    content: public override string ToString ();\n    parameters: []\n    return:\n      type: System.String\n      description: A string that represents the current <xref href=\"System.Data.SqlClient.SqlException\"></xref> object.<xref:System.String>.\n  overload: System.Data.SqlClient.SqlException.ToString*\n  exceptions: []\n  platform:\n  - net462\nreferences:\n- uid: System.Data.Common.DbException\n  isExternal: false\n  name: System.Data.Common.DbException\n- uid: System.ArgumentNullException\n  isExternal: true\n  name: System.ArgumentNullException\n- uid: System.Data.SqlClient.SqlException.Class\n  parent: System.Data.SqlClient.SqlException\n  isExternal: false\n  name: Class\n  nameWithType: SqlException.Class\n  fullName: System.Data.SqlClient.SqlException.Class\n- uid: System.Byte\n  parent: System\n  isExternal: true\n  name: Byte\n  nameWithType: Byte\n  fullName: System.Byte\n- uid: System.Data.SqlClient.SqlException.ClientConnectionId\n  parent: System.Data.SqlClient.SqlException\n  isExternal: false\n  name: ClientConnectionId\n  nameWithType: SqlException.ClientConnectionId\n  fullName: System.Data.SqlClient.SqlException.ClientConnectionId\n- uid: System.Guid\n  parent: System\n  isExternal: true\n  name: Guid\n  nameWithType: Guid\n  fullName: System.Guid\n- uid: System.Data.SqlClient.SqlException.Errors\n  parent: System.Data.SqlClient.SqlException\n  isExternal: false\n  name: Errors\n  nameWithType: SqlException.Errors\n  fullName: System.Data.SqlClient.SqlException.Errors\n- uid: System.Data.SqlClient.SqlErrorCollection\n  parent: System.Data.SqlClient\n  isExternal: false\n  name: SqlErrorCollection\n  nameWithType: SqlErrorCollection\n  fullName: System.Data.SqlClient.SqlErrorCollection\n- uid: System.Data.SqlClient.SqlException.GetObjectData(System.Runtime.Serialization.SerializationInfo,System.Runtime.Serialization.StreamingContext)\n  parent: System.Data.SqlClient.SqlException\n  isExternal: false\n  name: GetObjectData(SerializationInfo,StreamingContext)\n  nameWithType: SqlException.GetObjectData(SerializationInfo,StreamingContext)\n  fullName: System.Data.SqlClient.SqlException.GetObjectData(SerializationInfo,StreamingContext)\n- uid: System.Runtime.Serialization.SerializationInfo\n  parent: System.Runtime.Serialization\n  isExternal: false\n  name: SerializationInfo\n  nameWithType: SerializationInfo\n  fullName: System.Runtime.Serialization.SerializationInfo\n- uid: System.Runtime.Serialization.StreamingContext\n  parent: System.Runtime.Serialization\n  isExternal: true\n  name: StreamingContext\n  nameWithType: StreamingContext\n  fullName: System.Runtime.Serialization.StreamingContext\n- uid: System.Data.SqlClient.SqlException.LineNumber\n  parent: System.Data.SqlClient.SqlException\n  isExternal: false\n  name: LineNumber\n  nameWithType: SqlException.LineNumber\n  fullName: System.Data.SqlClient.SqlException.LineNumber\n- uid: System.Int32\n  parent: System\n  isExternal: true\n  name: Int32\n  nameWithType: Int32\n  fullName: System.Int32\n- uid: System.Data.SqlClient.SqlException.Number\n  parent: System.Data.SqlClient.SqlException\n  isExternal: false\n  name: Number\n  nameWithType: SqlException.Number\n  fullName: System.Data.SqlClient.SqlException.Number\n- uid: System.Data.SqlClient.SqlException.Procedure\n  parent: System.Data.SqlClient.SqlException\n  isExternal: false\n  name: Procedure\n  nameWithType: SqlException.Procedure\n  fullName: System.Data.SqlClient.SqlException.Procedure\n- uid: System.String\n  parent: System\n  isExternal: true\n  name: String\n  nameWithType: String\n  fullName: System.String\n- uid: System.Data.SqlClient.SqlException.Server\n  parent: System.Data.SqlClient.SqlException\n  isExternal: false\n  name: Server\n  nameWithType: SqlException.Server\n  fullName: System.Data.SqlClient.SqlException.Server\n- uid: System.Data.SqlClient.SqlException.Source\n  parent: System.Data.SqlClient.SqlException\n  isExternal: false\n  name: Source\n  nameWithType: SqlException.Source\n  fullName: System.Data.SqlClient.SqlException.Source\n- uid: System.Data.SqlClient.SqlException.State\n  parent: System.Data.SqlClient.SqlException\n  isExternal: false\n  name: State\n  nameWithType: SqlException.State\n  fullName: System.Data.SqlClient.SqlException.State\n- uid: System.Data.SqlClient.SqlException.ToString\n  parent: System.Data.SqlClient.SqlException\n  isExternal: false\n  name: ToString()\n  nameWithType: SqlException.ToString()\n  fullName: System.Data.SqlClient.SqlException.ToString()\n- uid: System.Data.SqlClient.SqlException.Class*\n  parent: System.Data.SqlClient.SqlException\n  isExternal: false\n  name: Class\n  nameWithType: SqlException.Class\n- uid: System.Data.SqlClient.SqlException.ClientConnectionId*\n  parent: System.Data.SqlClient.SqlException\n  isExternal: false\n  name: ClientConnectionId\n  nameWithType: SqlException.ClientConnectionId\n- uid: System.Data.SqlClient.SqlException.Errors*\n  parent: System.Data.SqlClient.SqlException\n  isExternal: false\n  name: Errors\n  nameWithType: SqlException.Errors\n- uid: System.Data.SqlClient.SqlException.GetObjectData*\n  parent: System.Data.SqlClient.SqlException\n  isExternal: false\n  name: GetObjectData\n  nameWithType: SqlException.GetObjectData\n- uid: System.Data.SqlClient.SqlException.LineNumber*\n  parent: System.Data.SqlClient.SqlException\n  isExternal: false\n  name: LineNumber\n  nameWithType: SqlException.LineNumber\n- uid: System.Data.SqlClient.SqlException.Number*\n  parent: System.Data.SqlClient.SqlException\n  isExternal: false\n  name: Number\n  nameWithType: SqlException.Number\n- uid: System.Data.SqlClient.SqlException.Procedure*\n  parent: System.Data.SqlClient.SqlException\n  isExternal: false\n  name: Procedure\n  nameWithType: SqlException.Procedure\n- uid: System.Data.SqlClient.SqlException.Server*\n  parent: System.Data.SqlClient.SqlException\n  isExternal: false\n  name: Server\n  nameWithType: SqlException.Server\n- uid: System.Data.SqlClient.SqlException.Source*\n  parent: System.Data.SqlClient.SqlException\n  isExternal: false\n  name: Source\n  nameWithType: SqlException.Source\n- uid: System.Data.SqlClient.SqlException.State*\n  parent: System.Data.SqlClient.SqlException\n  isExternal: false\n  name: State\n  nameWithType: SqlException.State\n- uid: System.Data.SqlClient.SqlException.ToString*\n  parent: System.Data.SqlClient.SqlException\n  isExternal: false\n  name: ToString\n  nameWithType: SqlException.ToString\n"}